### PR TITLE
Add source & dest to copy/move errors

### PR DESCRIPTION
**Pull request by sovainfo**
Still fighting with github to make it do what we want. Doesn't allow me to add changes to this pr.

### CODE REVIEW

**Comment by wilsonge** [comment]
If we aren't using the old strings anymore can we deprecate them please? Otherwise come Joomla 4.0 it's going to be horrendus to work out what strings we are and aren't using. Plus it makes it harder for end users

File: language/en-GB/en-GB.lib_joomla.ini [hunk 1]
```diff
@@ -260,6 +261,7 @@ JLIB_FILESYSTEM_ERROR_UPLOAD="JFile: :upload: %s"
 JLIB_FILESYSTEM_ERROR_WARNFS_ERR01="Warning: Failed to change file permissions!"
 JLIB_FILESYSTEM_ERROR_WARNFS_ERR02="Warning: Failed to move file!"
 JLIB_FILESYSTEM_ERROR_WARNFS_ERR03="Warning: File %s not uploaded for security reasons!"
+JLIB_FILESYSTEM_ERROR_WARNFS_ERR04="Warning: Failed to move file: %s => %s"
```
Comment (zero-24) [inline language/en-GB/en-GB.lib_joomla.ini]: @sovainfo can you use here also `%1s => %2s` :smile:

Comment (sovainfo) [inline language/en-GB/en-GB.lib_joomla.ini]: @zero-24 Thanks

**Comment by sovainfo** [comment]
Where is the documentation on deprecating messages?

**Comment by brianteeman** [comment]
There is no documentation for deprecating language strings. I did propose something but this was rejected.

@wilsonge How do you propose to deprecate the strings. The problem is that the stupid ancient way that TT are given strings to translate will not be able to show if a string is deprecated<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6644">issues.joomla.org/joomla-cms/6644</a>.</sub>

**Comment by sovainfo** [comment]
Considering there is no documentation on deprecating language string, this PR should be set to RTC.
See #6001 & #6002 

With special thanks to @davdebcom and @infograf768

**Comment by brianteeman** [comment]
Test instructions from @wilsonge From what I understand you need to install an extension when having not enough file permissions in the components (or some folder that's needed as described in the link in the description) to move files into the directory. Before the patch you'll just see a message "Copy file failed" and after you'll see a message stating what the to and from directory are.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6644">issues.joomla.org/joomla-cms/6644</a>.</sub>

**Comment by mikeveeckmans** [comment]
I have tested this item :white_check_mark: successfully on bd27a90832065e11df71ee032e8cc3a7c663aa7d<br /><br />TEST OK
(J3.5.1 , PHP 7)
made tmp folder locked and unwriteable.

screen shot 2016-04-07 at 04 42 59
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6644">issues.joomla.org/joomla-cms/6644</a>.</sub>

**Comment by infograf768** [comment]
Not sure I understand the messages:
```
Warning
Warning: Failed to move file: /Applications/MAMP/tmp/php/phptyMAxf => /Applications/MAMP/htdocs/trunkgitnew/tmp/jostag_plugin_for_2.5.zip
Error
Archive does not exist
Unable to find install package
```
what is `php/phptyMAxf` ?

Should not the message says
`Impossible to move the jostag_plugin_for_2.5.zip file to the /Applications/MAMP/htdocs/trunkgitnew/tmp/ folder`  ?

Also, I would use `to` instead of `=>` as it would be easier for rtl translations.

**Comment by brianteeman** [comment]
Valid point about using to

On 7 April 2016 at 11:22, infograf768 <notifications@github.com> wrote:

> Not sure I understand the messages:
>
> Warning
> Warning: Failed to move file: /Applications/MAMP/tmp/php/phptyMAxf => /Applications/MAMP/htdocs/trunkgitnew/tmp/jostag_plugin_for_2.5.zip
> Error
> Archive does not exist
> Unable to find install package
>
> what is php/phptyMAxf ?
>
> Should not the message says
> Impossible to move the jostag_plugin_for_2.5.zip file to the
> /Applications/MAMP/htdocs/trunkgitnew/tmp/ folder ?
>
> Also, I would use to instead of => as it would be easier for rtl
> translations.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6644#issuecomment-206800122>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by mbabker** [comment]
@infograf768 That path is the temporary path & name that whatever file you tried uploading gets sent to by PHP before the underlying code ultimately moves it into place.  JFile doesn't know what the name of the file you tried uploading, only what the source path it's located at is, so that's all the message can display when rendered from there.  To get a message like what you're looking for, it would need to be rendered by the method processing the upload as that will have access to the data extracted from the `$_FILES` superglobal (which contains that name).

**Comment by brianteeman** [comment]
Why create a new string  JLIB_FILESYSTEM_ERROR_WARNFS_ERR04 instead of updating  JLIB_FILESYSTEM_ERROR_WARNFS_ERR02<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6644">issues.joomla.org/joomla-cms/6644</a>.</sub>

**Comment by infograf768** [comment]
@mbabker 
I understand what you say. In that case I think that it is useless to display the source in the message as it is not understandable.
Why not simply use
`Warning: Failed to move file to /Applications/MAMP/htdocs/trunkgitnew/tmp/jostag_plugin_for_2.5.zip`

Btw, `Warning:` may be useless as it is stated in the Header of the message

**Comment by mbabker** [comment]
Source is only "not understandable" if you're moving something from a temporary location with a temporary file name.  JFile doesn't know whether it's been told to move something from the temporary location or told to move the index.php file to new.php.  So basically if you say hide it because some paths may look like gibberish then you have to not show the source path for all cases.

**Comment by sovainfo** [comment]
This PR corrects the mistake by core code of being too abstract. When something goes wrong it is not good enough to report something went wrong. You need to provide sufficient details in order to investigate the cause of whatever went wrong. Would be nice if Joomla was made so robust it would tell the cause, can't wait for that! In the mean time it would be helpful when Joomla provides sufficient details so you can investigate what the cause is. The fact core code is wrongly using the original messages doesn't any code using it is wrong. So, the core code must be changed to use a different message. Haven't found the new message, so it needed to be created. Obviously, any code making the same mistake can be corrected the same way. Correct usage of the message doesn't need any change!

**Comment by designbengel** [comment]
I have tested this item :white_check_mark: successfully on 17559154e33bd4113d40330df5364c34886f6534<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6644">issues.joomla.org/joomla-cms/6644</a>.</sub>

**Comment by mikeveeckmans** [comment]
I have tested this item :white_check_mark: successfully on 17559154e33bd4113d40330df5364c34886f6534<br /><br />test OK
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6644">issues.joomla.org/joomla-cms/6644</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6644">issues.joomla.org/joomla-cms/6644</a>.</sub>

**Pull request merged**